### PR TITLE
T23 - Voting contest

### PR DESCRIPTION
**Pull request by Harshini-Jesi**

### CODE REVIEW

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Program.cs [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
+         }
+      }
+      return maxChar;
+   }
+
+   /// <summary>Calculates the no.of repetition of the winner i.e.,maximum repeated char</summary>
+   /// <returns>The count i.e., votes of the winner</returns>
+   static int Votes (string input, char c) {
+      int count = 0;
+      foreach (char ch in input) {
```
Comment (MetaNaveen) [inline Program.cs:33]: Computationally costly. Requires n^2 iterations to solve the problem.

Try to use [Generic Collections](https://www.tutorialsteacher.com/csharp/csharp-collection) like Dictionary or List<(char, int)> to solve the problem more efficiently.

**Review by MetaNaveen — CHANGES_REQUESTED** [review]

File: Training.csproj [hunk 2]
```diff
@@ -4,6 +4,7 @@
     <OutputType>Exe</OutputType>
     <TargetFramework>net7.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
+    <GenerateDocumentationFile>True</GenerateDocumentationFile>
```
[the remark this answers is not in the document]
Comment (Harshini-Jesi) [reply Training.csproj:7]: Thank you!

File: Program.cs [hunk 3]
```diff
@@ -1,7 +1,31 @@
...
+   }
+
+   /// <summary>Finds the most repeated char and its no.of repetition in a string</summary>
+   /// <param name="input">The string in which each character represents the vote for a contestant</param>
+   /// <returns>The most repeated char i.e.,the winner and its no.of.repetition i.e.,the votes</returns>
+   static (char, int) Winner (string input) {
+      Dictionary<char, int> output = new ();
+      char maxChar = ' '; int maxCount = 0;
+      foreach (char c in input) {
+         if (output.TryGetValue (c, out int value)) output[c] = value + 1;
```
Comment (MetaNaveen) [inline Program.cs:21]: One-liner solution for the same:
output[c] = output.TryGetValue (c, out int value) ? ++value : 1;

**Review by MetaNaveen — CHANGES_REQUESTED** [review]
Review #2

File: Program.cs [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
-        }
-    }
+﻿/// <summary>This program takes a string as input with each character in it representing a vote for a contestant 
+/// and returns the winner with the most votes.</summary>
+class Program {
+   static void Main () {
+      Console.Write ("Enter a string: ");
+      string input = Console.ReadLine ().ToLower ();
+      do Console.WriteLine ($"{Winner (input)} => Winner and Votes");
+      while (!input.All (char.IsLetter) || input.Length == 0);
```
Comment (MetaNaveen) [inline Program.cs:8]: Runtime error. If I press Enter without giving any input, the code runs indefinitely.

**Review by MetaNaveen — CHANGES_REQUESTED** [review]
Review 3

**Review by MetaNaveen — APPROVED** [review]

**Comment by MetaNaveen** [comment]
Working fine.

**Pull request closed without merging**